feat(transport): allow defining multiple transport endpoints

NaddiNadja opened the pull request:
Added the `endpoint_name` parameter to `cijoe.run(...)`, which allows users to specify which endpoint they want to execute commands on. Endpoints are defined in the configuration file under the key `cijoe.transport` with user-defined names. If the `endpoint_name` parameter is not given, the command is executed on the first defined endpoint. If no endpoints are defined, the command is executed on the initiator.

Solves issue #75

karlowich reviewed (commented):
I propose that we rename `endpoint_name` to `endpoint`. To me this is cleaner, and makes for simpler code e.g. `for endpoint in endpoints:`

NaddiNadja commented:
> I propose that we rename `endpoint_name` to `endpoint`. To me this is cleaner, and makes for simpler code e.g. `for endpoint in endpoints:`

you're probably right, i have fixed this now, and moved the logic for finding the transport to a single method to reduce code redundancy

karlowich commented on src/cijoe/core/command.py:
```diff
@@ -196,7 +203,17 @@ def _run(self, cmd, cwd, env, transport):
 
-    def run(self, cmd, cwd=None, env={}):
+    def _get_transport(self, endpoint=None):
+        if not endpoint:
+            endpoint = next(iter(self.transports))
+        if endpoint not in self.transports:
```
Maybe this should be an `else if` since if the first is condition is true, this will always be true as well.
I initially misread the code, perhaps this change could make it a bit clearer.

NaddiNadja replied:
I know that if line 207 evaluates to true, then 209 will always evaluate to false (as the code is now), but because we change the value of `endpoint` (now renamed to `transport_name`) in line 208, I prefer to use another `if` over an `elif`. If we mess with line 208 in the future, line 209 is there to ensure that it doesn't break.
It's a constant-time lookup, so performance wise, it makes no difference, so I believe it's just a matter of preference.

safl commented:
> > I propose that we rename `endpoint_name` to `endpoint`. To me this is cleaner, and makes for simpler code e.g. `for endpoint in endpoints:`
> 
> you're probably right, i have fixed this now, and moved the logic for finding the transport to a single method to reduce code redundancy

In the issue #75, I referred to it as an "endpoint" / "transport endpoint," but this was likely not ideal, as it is referred to as "transport" everywhere else. I propose renaming it to ``transport_name: str`` to indicate that this is not an instance of the ``Transport`` class but rather a name specified in the configuration file.

safl reviewed (changes requested):
The commit message is now out of "tune" with the implementation and docs.

safl reviewed (approved):
LGTM.
